compose/src/main/kotlin/com/yourssu/handy/compose/Chip.kt

```diff
@@ -0,0 +1,111 @@
+import com.yourssu.handy.compose.foundation.*
+import com.yourssu.handy.compose.icons.HandyIcons
+import com.yourssu.handy.compose.icons.line.*
+
+enum class ChipSize(
+    internal val chipHeight: Dp
+) {
+    Large(32.dp),
+    Small(24.dp)
+}
+
+enum class ChipState {
+    Unselected,
+    Selected,
+    Disabled
+}
+
+@Composable
+fun Chip(
+    text: String = "",
+    modifier: Modifier = Modifier,
+    chipState: ChipState = ChipState.Unselected,
+    chipSize: ChipSize = ChipSize.Large,
+    leadingIcon: ImageVector? = HandyIcons.Line.InfoCircle,
+    trailingIcon: ImageVector? = HandyIcons.Line.Close,
+    chipOnClick: (() -> Unit)? = null,
+    leadingIconOnClick: (() -> Unit)? = null,
+    trailingIconOnClick: (() -> Unit)? = null
+) {
+
+    val backgroundColor = remember(chipState) {
+        when (chipState) {
+            ChipState.Unselected -> ColorGray100
+            ChipState.Selected -> ColorViolet100
+            ChipState.Disabled -> ColorGray50
+        }
+    }
+
+    val textColor = remember(chipState) {
+        when (chipState) {
+            ChipState.Unselected -> ColorGray700
+            ChipState.Selected -> ColorViolet500
+            ChipState.Disabled -> ColorGray300
+        }
+    }
+
+    val textStyle: HandyTextStyle =
+        if (chipState == ChipState.Selected) HandyTypography.B3Sb14 else HandyTypography.B3Rg14
+
+    Box(
+        modifier = modifier
+            .wrapContentWidth(unbounded = true)
+            .height(chipSize.chipHeight)
+            .background(backgroundColor, shape = RoundedCornerShape(16.dp))
+            .clickable(
+                enabled = chipState != ChipState.Disabled,
+                onClick = { chipOnClick?.invoke() })
+            .padding(horizontal = 12.dp),
+        contentAlignment = Alignment.Center
+    ) {
+        Row(
+            verticalAlignment = Alignment.CenterVertically,
+            horizontalArrangement = Arrangement.spacedBy(4.dp)
+        ) {
+            if (leadingIcon != null)
+                Icon(
+                    leadingIcon,
+                    tint = textColor,
+                    iconSize = IconSize.XS,
+                    modifier = Modifier.clickable(
+                        enabled = chipState != ChipState.Disabled,
+                        onClick = { leadingIconOnClick?.invoke() })
+                )
+
+            Text(
+                text = if (text.length < 10) text else text.take(9) + "...",
+                color = textColor,
+                fontSize = 14.dp,
+                style = textStyle
+            )
+
+            if (trailingIcon != null)
+                Icon(
+                    trailingIcon,
+                    tint = textColor,
+                    iconSize = IconSize.XS,
+                    modifier = Modifier.clickable(
+                        enabled = chipState != ChipState.Disabled,
+                        onClick = { trailingIconOnClick?.invoke() })
+                )
+        }
```

Comment:
```suggestion
            trailingIcon?.let{
                Icon(
                    trailingIcon,
                    tint = textColor,
                    iconSize = IconSize.XS,
                    modifier = Modifier.clickable(
                        enabled = chipState != ChipState.Disabled,
                        onClick = { trailingIconOnClick?.invoke() })
                )
            }
```

이건 수정안하셔도 되긴한데 전 이런 표현이 조금 더 kotlin스러운것같아 선호해요

Reply:
확실히 가독성도 더 좋은 것 같습니다!